python/GangaLHCb/Lib/RTHandlers/GaudiExecRTHandlers.py

```diff
@@ -330,10 +331,19 @@ def uploadLocalFile(job, namePattern, localDir, should_del=True):
     """
 
     new_df = DiracFile(namePattern, localDir=localDir)
-    random_SE = random.choice(getConfig('DIRAC')['allDiracSE'])
+    trySEs = getConfig('DIRAC')['allDiracSE']
+    random.shuffle(trySEs)
     new_lfn = os.path.join(getInputFileDir(job), namePattern)
-    returnable = new_df.put(force=True, uploadSE=random_SE, lfn=new_lfn)[0]
-
+    for SE in trySEs:
+        try:
+            returnable = new_df.put(force=True, uploadSE=SE, lfn=new_lfn)[0]
+        except:
+            raise GangaException("Upload of LFN %s to SE %s failed" % (new_lfn, SE)) 
+        if new_df.failureReason == '':
+            break
+        else:
+            logger.warning("Unable to upload LFN %s to SE %s. Trying another SE." % (new_lfn, SE))
+            new_df.failureReason = ''
```

Comment:
@mesmith75 IF the upload fails (to any SE), i.e. it just runs to the end of the loop and give `failureReason` different from the empty string for all SEs, then this code will not raise an exception. Is that the intention?